fog/view/enclave/impl/src/oblivious_utils.rs

```diff
@@ -0,0 +1,508 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
```

Comment:
Note: reviewed the diff of this with #2252 since #2252 seems to be bringing in the majority of this logic.